core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -572,7 +600,7 @@ $(document).ready(function () {
                                 var row1 = createRowMetadataForColumn(
                                     columnKey, taskMetricsResponse[columnKey], 4);
                                 var row2 = createRowMetadataForColumn(
-                                    "shuffleWriteTime", taskMetricsResponse[columnKey], 21);
+                                    "shuffleWriteTime", taskMetricsResponse[columnKey], 23);
```

Comment:
Hi, @AngersZhuuuu . If possible, could you try to avoid this change?

Reply:
> If possible, could you try to a

Since add new column in Additional Metric, the column index is changed. So need to do this

To be honest , I spend a lot of time to make sure the column change when add new column, since the spark web ui code changed a lot, since origin pr in `stagepage.js` have not do things like add column.

If it's ok, this pr can be a demo for new develope of spark web UI

Reply:
That's the exact reason why I asked that. I've seen several UI patches like this. :)
> Since add new column in Additional Metric, the column index is changed. So need to do this

There exists lots of similar commits already for new develop of Spark web UI. Always, the commits(already reviewed and merged) are the best demo for developers.
> If it's ok, this pr can be a demo for new develope of spark web UI

Reply:
> There exists lots of similar commits already for new develop of Spark web UI. Always, the commits(already reviewed and merged) are the best demo for developers.

Each page's JS script isn't the same, and I have find that for stage page, don't have merged pr about add new additional column.

Reply:
@AngersZhuuuu I found following issues.

1. When a job performs shuffle, StagePage for shuffle write stage has a problem.
Even if you visit StagePage where the corresponding stage performs shuffle write and check `Shuffle Write Time`,
the metrics is not shown.
<img width="1426" alt="スクリーンショット 2020-04-08 16 14 30" src="https://user-images.githubusercontent.com/4736016/78756030-136a5a80-79b5-11ea-90f3-917e31839588.png">

2. Web UI will be hung-up
When you visit a StagePage where the corresponding stage performs shuffle read and reload, UI will be hung-up.
<img width="1438" alt="hungup" src="https://user-images.githubusercontent.com/4736016/78756200-5f1d0400-79b5-11ea-967a-6992fd4b5a4f.png">


Those issues are caused because column indices are not changed properly.
To avoid such issues, please keep it simple. I believe column indices need not to be changed.